src/net.h

```diff
@@ -1095,7 +1095,7 @@ class CNode
     CService addrLocal GUARDED_BY(cs_addrLocal);
     mutable RecursiveMutex cs_addrLocal;
 
-    //! Whether this peer connected via our Tor onion service.
+    //! Whether this peer is an inbound onion, e.g. connected via our Tor onion service.
     const bool m_inbound_onion{false};
```

Comment:
yocto-nit (not directly related to this PR though, feel free to ignore): not sure if we have any guidelines (or dominant opinions) on this subject, but isn't it overkill to have two initializations for a class member: once via in-class initialization and once via initializer list in the ctor? (And yes, I'm aware that initializer list have higher priority in C++). As reader of the class interface I'd not assume an in-class initialization is overruled in the ctor. Also, without the in-class initialization the compiler has the chance to throw an error if the const member was forgot to be initializer.

Reply:
Hm, good eye.

-> In this case, ISTM we could remove the default `false` argument in the constructor declaration

-> As to your last point, it suggests the default member `m_inbound_onion` initializer not set a value, since we are doing it dynamically in the ctor. It's true that I see a build error in this case, but not without the change in `net.h:1099`
```diff
/src/net.cpp
-    nMyStartingHeight(nMyStartingHeightIn),
-    m_inbound_onion(inbound_onion && conn_type_in == ConnectionType::INBOUND)
+    nMyStartingHeight(nMyStartingHeightIn)

/src/net.h
     //! Whether this peer is an inbound onion, e.g. connected via our Tor onion service.
-    const bool m_inbound_onion{false};
+    const bool m_inbound_onion;
```
```
net.cpp:2947:1: error: uninitialized const member in ‘const bool’ [-fpermissive]
 2947 | CNode::CNode(NodeId idIn, ServiceFlags nLocalServicesIn, int nMyStartingHeightIn, SOCKET hSocketIn, const CAddress& addrIn, uint64_t nKeyedNetGroupIn, uint64_t nLocalHostNonceIn, const CAddress& addrBindIn, const std::string& addrNameIn, ConnectionType conn_type_in, bool inbound_onion)
      | ^~~~~
In file included from net.cpp:10:
./net.h:1099:16: note: ‘const bool CNode::m_inbound_onion’ should be initialized
 1099 |     const bool m_inbound_onion;
      |                ^~~~~~~~~~~~~~~
```
I don't mind pulling in a commit from your branch if you'd like to adjust this.

-----

More generally, from what I can gather, the ideas behind in-class default member initializers added in C++11 were (a) to allow a non-static data member to be initialized where it is declared, e.g. in its class. A constructor can then use the initializer when run-time initialization is needed, which can then (b) tidy up the code up by avoiding doing it in the member initializer list of the constructor. This provides extra benefits in classes with multiple constructors.

I guess one can see the default member initializer as the general default, particularly useful for constant initializers, that can be overridden by any specific ones in member initializer lists in the constructors.

Some resources (send me any good ones you might suggest; my understanding on this is evolving):
- https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#Rc-in-class-initializer
- https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#c49-prefer-initialization-to-assignment-in-constructors
- https://en.cppreference.com/w/cpp/language/data_members#member-initialization
- https://www.stroustrup.com/C++11FAQ.html#member-init

Reply:
(Or you could open a separate PR, for as you mentioned, it's not directly related to this change. Either way SGTM.)

Reply:
> not sure if we have any guidelines (or dominant opinions) on this subject, but isn't it overkill to have two initializations for a class member: once via in-class initialization and once via initializer list in the ctor? (And yes, I'm aware that initializer lists have higher priority in C++). As reader of the class interface I'd not assume an in-class initialization is overruled in the ctor. Also, without the in-class initialization the compiler has the chance to throw an error if the const member was forgot to be initialized in the ctor.

@theStack: done in #21167